lib/Parse/ParseDecl.cpp

```diff
@@ -356,9 +356,11 @@ void Parser::parseSourceFileViaASTGen(
                                        appendToVector);
 
     // Spin the C++ parser to the end; we won't be using it.
+    DiagnosticTransaction noDiag(Context.Diags);
```

Comment:
This avoids `consumeToken()` emits lexer diagnostics e.g. "editor placeholder in source file"

Reply:
Ah, nice.